osimpipeline/vital_tasks.py

```diff
@@ -542,7 +507,327 @@ def __init__(self, trial, ik_setup_task):
+                'ik', '%s_%s_ik_solution.mot' % (self.study.name, trial.id))
+        else:
+            raise Exception("TaskCMCSetup: %s is not a valid kinematics task "
+                "source, please choose 'rrakin' or 'ik'." % des_kinematics)
+        
+        # Set control constraints path (if any)
+        if self.control_constraints:
+            self.control_constraints_fpath = 'TODO'
+        else:
+            self.control_constraints_fpath = ''
 
+        # TODO
+        # self.coord_act_fpath
 
+        # Fill out external_loads.xml template and copy over to results 
+        # directory
+        print "debug1: " + self.results_extloads_fpath
+        self.copy_over_external_loads()
 
+        # Fill out tasks.xml template and copy over to results directory
+        self.copy_over_tasks()
+
+    # Override derived action method since different desired kinematics
+    # may be specified 
+    def fill_external_loads_template(self, file_dep, target):
+        with open(file_dep[0]) as ft:
+            content = ft.read()
+            content = content.replace('@STUDYNAME@', self.study.name)
+            content = content.replace('@NAME@', self.trial.id)
+            content = content.replace('@DESKINEMATICS@', 
+                self.des_kinematics_fpath)
+
+        with open(target[0], 'w') as f:
+            f.write(content)
+
+    def fill_setup_template(self, file_dep, target, 
+                            init_time=None, final_time=None):
+        with open(file_dep[0]) as ft:
+            content = ft.read()
+            content = content.replace('@STUDYNAME@', self.study.name)
+            content = content.replace('@NAME@', self.trial.id)
+            content = content.replace('@MODEL@', 
+                                      self.adjusted_model_fpath)
+            content = content.replace('@INIT_TIME@', '%.4f' % init_time)
+            content = content.replace('@FINAL_TIME@', '%.4f' % final_time)
+            force_set_files = '%s %s' % (self.cmc_act_fpath,
+                self.res_act_fpath)
+            content = content.replace('@FORCESETFILES@', force_set_files)
+            content = content.replace('@DESKINEMATICS@',
+                self.des_kinematics_fpath)
+            content = content.replace('@CONTROLCONSTRAINTS@', 
+                self.control_constraints_fpath)
+            content = content.replace('@EXTLOADS@', 
+                self.results_extloads_fpath)
+            content = content.replace('@TASKS@', self.results_tasks_fpath)
+        
+        with open(target[0], 'w') as f:
+            f.write(content)
+
+class TaskCMC(task.ToolTask):
+    REGISTRY = []
+    def __init__(self, trial, cmc_setup_task, cycle=None):
+        super(TaskCMC, self).__init__('cmc', trial, cycle=cycle)
+        self.doc = "Run OpenSim's Computed Muscle Control tool."
+        self.des_kinetics_fpath = \
+            '%s/expdata/ground_reaction_orig.mot' % trial.results_exp_path
```

Comment:
Oh...right. I have a task that renames the columns in the MOT file. Others could also have a task that filters the data. It could be nice to allow people to have intermediate processing steps for the MOT file. The way I have it set up for `mrsdeviceopt` is that `ground_reaction_orig.mot` is what's copied over from the data directory, and `ground_reaction.mot` is what's used by the tools, and there's an intermediate task that creates `ground_reaction.mot` from `ground_reaction_orig.mot`. I'm not sure how to handle this generically. For simplicity, you can just use the same file everywhere for now.